src/mtev_cluster.c

```diff
@@ -801,19 +806,22 @@ mtev_cluster_set_self(uuid_t id) {
        my_id_str[UUID_STR_LEN+1];
   uuid_copy(my_cluster_id, id);
   uuid_unparse_lower(my_cluster_id, my_id_str);
-  if(!mtev_conf_get_stringbuf(MTEV_CONF_ROOT, "//clusters/@my_id",
-                              old_uuid, sizeof(old_uuid)) ||
-     strcmp(old_uuid, my_id_str)) {
+  int found_id = mtev_conf_get_stringbuf(MTEV_CONF_ROOT, "//clusters/@my_id",
+                                         old_uuid, sizeof(old_uuid));
+  if(!found_id || strcmp(old_uuid, my_id_str) != 0) {
```

Comment:
I don't understand this change?  It seems to not change behavior, but make it look less like predicates elsewhere in libmtev.

Reply:
This is related to a discussion in #syseng a few days ago: https://circonus.slack.com/archives/C0J5RAKDJ/p1518795659000234
I found this critical if statement very intransparent:
1. Side effects inside an if condition are confusing. First do something then check for errors.
2. The return codes of functions like mtev_conf_get_stringbuf, and strcmp, etc. are inconsistent. Does a non-zero value indicate success or error? No way to find this out without checking the docs/implementation, just "knowing" (which I do not). Here I named the results and makes the check more explicit.
For me the committed version is much more readable.

Reply:
Consistency is king. It is standard practice throughout mtev code to use `strcmp`'s output as a boolean indicating nonmatch.  Also, almost everywhere in the code we use the boolean output of an mtev_conf_get function as success criteria.

This change deforms the code from its normal. I'm not supportive of this change.  It doesn't fix a bug and moves the code away from what is normal throughout the code base.

Reply:
OK. Reverting.